File: 프로그래머스/unrated/181906. 접두사인지 확인하기/접두사인지 확인하기.java
class Solution {
    public int solution(String my_string, String is_prefix) {
        
        if(my_string.charAt(0) != is_prefix.charAt(0) || is_prefix.length() > my_string.length()) return 0;
        
        String slicedString = my_string.substring(0, is_prefix.length());
        
        return slicedString.matches(is_prefix) ? 1 : 0;
    }
}

File: 프로그래머스/unrated/181906. 접두사인지 확인하기/접두사인지 확인하기.js
function solution(my_string, is_prefix) {
    
    if(my_string[0] !== is_prefix[0] || is_prefix.length > my_string.length) return 0;
    
    const length = is_prefix.length;
    
    const slicedString = my_string.slice(0, length);
   
    return slicedString === is_prefix ? 1 : 0;
}